### PR TITLE
Add acceptance tests for checking that share with group and user with same name works

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add acceptance tests for checking that the shares with user and group with same name works.

## Related Issue

- Checking bug in #35488 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by phil-davis — COMMENTED** [review]
Please also make some API test scenarios with the sharing API.
Add scenario(s) for when the group name has different case from the user-id. e.g. group `User1` as well as group `user`
It will be good to also have tests that demonstrate how the API behaves.

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroupsEdgeCases.feature [hunk 1]
```diff
@@ -32,3 +32,42 @@ Feature: Sharing files and folders with internal groups
...
+      | user1     |
+    And user "user1" has been added to group "user1"
+    And user "user3" has logged in using the webUI
+    When the user shares folder "simple-folder" with group "user1" using the webUI
+    And the user shares folder "simple-folder" with user "User One" using the webUI
+    # Remove this after issue is fixed
+    Then a notification should be displayed on the webUI with the text "Path already shared with this user"
+    When the user re-logs in as "user1" using the webUI
+    Then folder "simple-folder" should be marked as shared with "user1" by "User Three" on the webUI
+    # Uncomment this line after issue is fixed# And folder "simple-folder" should be marked as shared with "User One" by "User Three" on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroupsEdgeCases.feature]: Needs a new line.

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 2]
```diff
@@ -669,6 +669,7 @@ public function isFieldInResponse($field, $contentExpected, $expectSuccess = tru
 					)
 				) . " 00:00:00";
 		}
+		var_dump($data);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Sharing.php]: delete debug code

File: tests/acceptance/features/bootstrap/OCSContext.php [hunk 3]
```diff
@@ -349,10 +349,11 @@ public function userSendsRequestToTheseEndpointsWithBodyUsingPassword($user, $me
 	 * @return void
 	 */
 	public function theOCSStatusCodeShouldBe($statusCode, $message = "") {
+		var_dump($this->featureContext->getResponseXml());
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OCSContext.php]: delete debug code

**Review by phil-davis — COMMENTED** [review]
Tests look code - proves that this is a Sharing API bug.

**Review by phil-davis — APPROVED** [review]

**Comment by dpakach** [comment]
@ownclouders rebase

**Pull request merged**